### PR TITLE
feat(meta): support `risectl throttle` for table with source as well

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support throttle for table with source.

1. Need to find source ID, perhaps via meta dashboard / grafana metrics.
2. `./risingwave ctl throttle source <id> <ratelimit, e.g. 1>`

Screenshot 2024-02-26 at 3 55 24 PM

source id is 1010 here.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by fuyufjh — APPROVED** [review]

**Review by tabVersion — APPROVED** [review]

File: src/meta/src/manager/catalog/fragment.rs [hunk 1]
```diff
@@ -869,9 +869,14 @@ impl FragmentManager {
...
         for (table_id, table_fragments) in map.iter() {
             for fragment in table_fragments.fragments.values() {
+                // CREATE SOURCE
                 if (fragment.get_fragment_type_mask() & FragmentTypeFlag::Source as u32) != 0 {
                     table_id_to_apply.insert(*table_id);
                 }
+                // CREATE TABLE WITH (connector='...')
+                if (fragment.get_fragment_type_mask() & FragmentTypeFlag::Mview as u32) != 0 {
+                    table_id_to_apply.insert(*table_id);
+                }
```
Comment (BugenZhao) [inline src/meta/src/manager/catalog/fragment.rs:879]: Do you mean that we don't have a fragment with `Flag::Source` in a table job with `CREATE TABLE WITH (connector='...')`?

Comment (BugenZhao) [reply src/meta/src/manager/catalog/fragment.rs:879]: The code here looks weird to me. https://github.com/risingwavelabs/risingwave/blob/468f7ec14f7e53d7c4d4af965eace50bf1d2fde7/src/meta/src/manager/catalog/fragment.rs#L877-L882

says that "source_id not found", but we don't actually check the source id at all in the code above. It appears to just filter out all streaming jobs without a `Source` fragment (which means jobs on MVs).

**Comment by kwannoel** [comment]
See: https://github.com/risingwavelabs/risingwave/pull/15255#discussion_r1502208829.

`Flag::Source` should be present, we still have a source fragment.

File: src/meta/src/manager/catalog/fragment.rs (hunk 1 above)
Comment (kwannoel) [reply src/meta/src/manager/catalog/fragment.rs:879]: cc @tabVersion can you PTAL at @BugenZhao 's remarks?

**Pull request closed without merging**